modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_rule_builder.js

```diff
@@ -240,7 +242,32 @@ AUI.add(
 						instance._ruleClasses.set('fields', instance.getFields());
 						instance._ruleClasses.set('pages', instance.getPages());
 
-						instance._ruleClasses.render(rule);
+						if (!instance.ruleDraftEmpty(ruleDraft)) {
+							instance._ruleClasses.render(ruleDraft);
+						}
+						else {
+							instance._ruleClasses.render(rule);
+						}
+					},
+
+					ruleDraftEmpty: function(ruleDraft) {
+						var ruleDraftIsEmpty = true;
+
+						for (var i in ruleDraft) {
+							if (ruleDraft.hasOwnProperty(i)) {
+								ruleDraftIsEmpty = false;
+								break;
+							}
+						}
+						if (ruleDraft && ruleDraft.conditions) {
+							for (var j = 0; j < ruleDraft.conditions.length; j++) {
```

Comment:
can you use an enhanced loop? for (var condition in ruleDraft.conditions) { ... }